controllers/userController.js

```diff
@@ -7,23 +11,52 @@ const getUsers = (req, res) => {
     res.json(users)
 };
 
-const createUser = (req, res) => {
-    const name = req.body.name;
-    const email = req.body.email;
-    const user = {
+const createUser = async (req, res) => {
+    try {
+        const salt = await bcrypt.genSalt();
+        const hashPassword = await bcrypt.hash(req.body.password, salt);
+        const name = req.body.name;
+        const email = req.body.email;
+        const password = hashPassword;
+        const user = {
         id: uuidv4(),
         email,
         name,
+        password
     };
-    if (!user.name) {return res.status(400).send('A name should be provided')}
+    if (!user.name) {res.status(400).send('A name should be provided')}
     users.push(user);
-    res.status(201).json(user);
+    res.status(201);
+
+    // Send JSON WEB TOKEN
+    const token = JWT.sign(user, process.env.SECRET_KEY)
```

Comment:
Looks to me like you are using the entire user object as the payload for the token. Keep in mind that JWT is encoded not encrypted, and the user's password even if hashed, is in that user object. You never want to return the user's password to the client in any form, encrypted or not.

To validate what I said about JWT not being encrypted, generate one and paste it in the encoded part of https://jwt.io/ to see the payload.

Reply:
I did this and got all the user info, name, email and the hashed password as the payload, should i just send only the name back to the client